knack/config.py

```diff
@@ -109,9 +109,9 @@ def sections(self):
 
     def items(self, section):
         import re
-        pattern = self.env_var_name(section, '.+')
-        candidates = [(k.split('_')[-1], os.environ[k], k) for k in os.environ if re.match(pattern, k)]
-        result = {c[0]: c for c in candidates}
+        pattern = self.env_var_name(section, '[0-9A-Z][0-9A-Z_]*')
+        candidates = [(k.replace(self.env_var_name(section, ''), ''), os.environ[k], k) for k in os.environ if re.match(pattern, k)]
```

Comment:
`replace` can cause unwanted effect if `self.env_var_name(section, '')` somehow appears in the middle of the env var. [`removeprefix`](https://docs.python.org/3/library/stdtypes.html#str.removeprefix) is a better choice but it is only available since Python 3.9.

Better to use [grouping](https://docs.python.org/3/howto/regex.html#grouping) of regex.